example/linux/gw-example/main.c

```diff
@@ -101,11 +101,11 @@ static bool MQTT_publish(char * topic, uint8_t * payload, size_t payload_size, b
 static void * publish_thread(void * unused)
 {
     message_to_publish_t * message;
-    pthread_mutex_lock(&m_queue_mutex);
     while (true)
     {
         if (m_queue_empty)
         {
+            pthread_mutex_lock(&m_queue_mutex);
```

Comment:
It seems ok, but we should add a lock unlock when we release the message. Ligne 141 lock, 148 unlock.